### PR TITLE
[native]Add multi write driver support

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Add table write merge node and operator creation support in Presto to
Velox query plan conversion to support the query shape with table write
merge session option on;
Add hive bucket function support in local partition node in Presto to Velox
query plan conversion to support multiple table write drivers;
Turn on table write merge and other multi table write driver session
properties in native java e2e test.

Also advance velox version

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ TEST_F(PlanNodeSerdeTest, partitionAndSerializeNode) {
                   .values(data_, true)
                   .addNode(addPartitionAndSerializeNode(
                       4, false, reverseColumns(asRowType(data_[0]->type()))))
-                  .localPartition({})
+                  .localPartition(std::vector<std::string>{})
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:74]: Are these changes needed?

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 2]
```diff
@@ -1223,29 +1223,60 @@ core::PlanNodePtr VeloxQueryPlanConverterBase::toVeloxQueryPlan(
         sourceNodes[i]);
   }
 
-  if (isHashPartition(node)) {
+  if (type == core::LocalPartitionNode::Type::kGather) {
+    return core::LocalPartitionNode::gather(node->id, std::move(sourceNodes));
+  }
+
+  auto connectorHandle =
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1230]: This method is very long. Consider extracting some helper functions.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 3]
```diff
@@ -1031,7 +1031,9 @@ private Session buildSessionForTableWrite()
     {
         // TODO: enable this after column stats collection is enabled.
         return Session.builder(getSession())
-                .setSystemProperty("table_writer_merge_operator_enabled", "false")
+                .setSystemProperty("table_writer_merge_operator_enabled", "true")
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:1034]: Nice.

**Review by mbasmanova — APPROVED** [review]
Looks good. Next time, please, put advance velox changes into a separate commit.

File: presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp (hunk 1 above)
Comment (xiaoxmeng) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:74]: Yes, there is ambiguity if we just {} as localPartition could take a shared ptr of BucketedFunctionProperty as input

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1289]: Update!

File: presto-native-execution/presto_cpp/main/types/CMakeLists.txt [hunk 4]
```diff
@@ -14,6 +14,8 @@ add_library(
   TypeSignatureTypeConverter.cpp antlr/TypeSignatureLexer.cpp
   antlr/TypeSignatureParser.cpp)
 
+include_directories(/opt/homebrew/opt/protobuf/include/)
```
Comment (majetideepak) [inline presto-native-execution/presto_cpp/main/types/CMakeLists.txt:17]: @xiaoxmeng is this needed?
